test/src/basic/fileformats.cpp

```diff
@@ -671,6 +671,37 @@ void describeGAFormat(const std::string name, GraphIO::AttrReaderFunc readerGA,
 	});
 }
 
+/**
+ * Tests reading a DOT clustergraph, using a simplified version of:
+ * https://graphviz.gitlab.io/_pages/Gallery/directed/cluster.html
+ */
+void describeDOTwithClusters()
+{
+	describe("DOT with subgraphs as clusters", []() {
+		it("reads a cluster graph", []() {
+			const std::string filename = "fileformats/dot/valid/cluster";
+
+			const ResourceFile* file = ResourceFile::get(filename);
+			std::stringstream is{file->data()};
+
+			Graph G;
+			ClusterGraph CG(G);
+			ClusterGraphAttributes CA(CG);
+
+			const bool readStatus = GraphIO::readDOT(CA, CG, G, is);
```

Comment:
You don't need `ClusterGraphAttributes` here, so skip the definition and use `readDOT(CA, G, is)`.